src/org/labkey/remoteapi/core/SaveModulePropertiesCommand.java

```diff
@@ -0,0 +1,63 @@
+package org.labkey.remoteapi.core;
+
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.labkey.remoteapi.CommandException;
+import org.labkey.remoteapi.CommandResponse;
+import org.labkey.remoteapi.Connection;
+import org.labkey.remoteapi.PostCommand;
+import org.labkey.remoteapi.security.GetContainersCommand;
+import org.labkey.remoteapi.security.GetContainersResponse;
+import org.labkey.test.params.ModuleProperty;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class SaveModulePropertiesCommand extends PostCommand<CommandResponse>
+{
+    private final List<ModuleProperty> _moduleProperties;
+    private final Map<String, String> _containerIds = new HashMap<>();
+
+    public SaveModulePropertiesCommand(List<? extends ModuleProperty> moduleProperties)
+    {
+        super("core", "saveModuleProperties");
+        _moduleProperties = Collections.unmodifiableList(moduleProperties);
+    }
+
+    @Override
+    public CommandResponse execute(Connection connection, String folderPath) throws IOException, CommandException
+    {
+        for (ModuleProperty prop : _moduleProperties)
+        {
+            String containerPath = prop.getContainerPath();
+            if (!_containerIds.containsKey(containerPath))
+            {
+                GetContainersResponse response = new GetContainersCommand().execute(connection, containerPath);
+                _containerIds.put(containerPath, response.getContainerId());
+            }
+        }
+        return super.execute(connection, folderPath);
+    }
+
+    @Override
+    public JSONObject getJsonObject()
+    {
+        JSONObject json = new JSONObject();
+        JSONArray properties = new JSONArray();
+        for (ModuleProperty prop : _moduleProperties)
+        {
+            JSONObject propJson = new JSONObject();
+            propJson.put("userId", 0);
```

Comment:
Do you understand the purpose of this "userId"... the action blows up if it's missing or invalid (not a real user), but then it otherwise ignores this value.

Reply:
I just saw it in the POST through the UI and matched that.
Looks like it's a vestigial remnant that snuck by a couple of old refactors.

Reply:
This comment implies that there was once a intention to allow user-specific module properties: https://github.com/LabKey/platform/blob/develop/core/webapp/ModulePropertiesAdminPanel.js#L237

Reply:
Cool, thanks. This doesn't hold up your merge, of course. I'll check in with Ben & Josh to verify we should rip out the now pointless server-side check.